Automattic/wp-calypso
# Pattern Library: Implement home page search results page

Fixes https://github.com/Automattic/dotcom-forge/issues/5953

## Proposed Changes

Previously, searching from the `/patterns` landing page didn't look right. The search results page is supposed to look like a category page, but instead, we just displayed the search results below the first `CategoryGallery`. This PR fixes that!

Searching from `/patterns` takes users to a URL like, e.g., `/patterns?s=contact`. This means that technically, they are still on the same `/patterns` page. We still want `/patterns?s=contact` to look like a category page, which is why I combined `patterns/pages/home` and `patterns/pages/category` into a single page in this PR.

Me and @nightnei discussed potential alternative approaches for implementing this, but decided to move forward with this approach for now and revisit it in https://github.com/Automattic/dotcom-forge/issues/6110

<img src="https://github.com/Automattic/wp-calypso/assets/1101677/ef6969ef-4bbb-4d86-ba5a-e73be5c592f1" width="350" />

## Testing Instructions



1. Navigate to `/patterns`
2. Search for `contact`
3. Ensure that 15 results are displayed
4. Click the `Discover` link in the category pill nav
5. Ensure that you're taken back to `/patterns` and that the search field is cleared
6. Click `About`
7. Search for `contact`
8. Ensure that 1 pattern is displayed
9. Click the `Blog Posts` link in the category pill nav
10. Ensure that you're taken back to `/patterns/posts` and that the search field is cleared
